src/Manager.php

```diff
@@ -570,8 +575,15 @@ private function executeRule(string $user, Item $item, array $params): bool
             ->execute($user, $item, $params);
     }
 
+    /**
+     * @throws ItemAlreadyExistsException
+     */
     private function addItem(Item $item): void
     {
+        if ($this->itemsStorage->get($item->getName()) !== null) {
```

Comment:
You've added `exists()` method to `ItemsStorage` interface and not using it. Either interface method should be removed or used.

Reply:
Fixed